### DISCUSSION TITLE
Using suffixAction() to create a new record instead of the createOptionForm()?

### DISCUSSION THREAD

**Discussion by sasori13L** [filamentphp/filament#12369] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.0

### How can we help you?

Is there a way to use the suffix action to create a new record to a select relationship database, instead of the createOptionForm?
or if there is a way to change the icon of the createOptionForm() and make it slideOver rather than modal.

![image](https://github.com/filamentphp/filament/assets/16425810/b34b15c1-5947-4ac4-9601-cc8a84f908d6)

```
Forms\Components\Select::make('guest_id')
  ->required()
  ->label('Guest Name')
  ->options(Guest::all()->pluck('name', 'id'))
  ->createOptionForm(fn(Form $form) => RoomResource::form($form))
  ->suffixAction(
      Forms\Components\Actions\Action::make('new_guest')
          ->slideOver()
          ->icon('heroicon-m-user-plus')
          ->iconButton()
          ->form(fn(Forms\Form $form) => GuestResource::form($form)))
          ->after(function ($state) {

          })
  ->searchable()
```

**Comment by dissto** [comment, marked as answer]
```php
->createOptionAction(
        fn (Action $action) => $action->icon('heroicon-m-user-plus')
    )
```

https://filamentphp.com/docs/3.x/forms/fields/select#customizing-the-select-action-objects
